zerver/lib/upload.py

```diff
@@ -844,6 +875,9 @@ def upload_avatar_image(user_file: File, acting_user_profile: UserProfile,
     upload_backend.upload_avatar_image(user_file, acting_user_profile,
                                        target_user_profile, content_type=content_type)
 
+def upload_avatar_image_from_url(url: str, user_profile: UserProfile) -> None:
```

Comment:
The image-fetching logic is independent of the upload backend used, so I think the function `upload_avatar_image_from_url` should download the image and then call `upload_avatar_image`? Eliminating the need to add the extra backend methods.

Also, instead of using an actual file, something like `BytesIO` could be the best solution? That's what `sync_avatar_from_ldap` uses too. https://stackoverflow.com/a/44672691